backend/app/src/main/java/com/moh/phlat/backend/service/SpecificationService.java

```diff
@@ -0,0 +1,22 @@
+package com.moh.phlat.backend.service;
+
+import java.util.List;
+
+import org.springframework.data.jpa.domain.Specification;
+
+import com.moh.phlat.backend.model.ProcessData;
+import com.moh.phlat.backend.model.SourceData;
+
+public interface SpecificationService {
```

Comment:
```suggestion
public interface ProcessDataFilterSpecification{
```
Also could we create separate specification classes for source data and process data. It seems we are mixing the two and give some specific name like suggested above.

Reply:
Split into two files.